### PR TITLE
chore: add system tests

### PR DESCRIPTION
**Pull request by nolanmar511**
Set up initial system tests for pprof-nodejs. 

Tested:
Looked at system test output logs to confirm expected version of Node is used. 
Locally ran validate that v8-canary test will run once kokoro configs are setup for it.

System tests which test the SourceMapper will be in a follow-up change.

### CODE REVIEW

**Comment by nolanmar511** [comment]
I've gotten the test working when I only collect a time profile. 
I'm getting a SegFault on the kokoro VM when I try to collect and save a heap profile. 

I used the [segfault-handler](https://www.npmjs.com/package/segfault-handler) npm module, and I got this trace:
```
            Tue Mar  5 16:34:37 PST 2019: + node --trace-warnings build/src/busybench.js 30
            Tue Mar  5 16:34:39 PST 2019: PID 7718 received SIGSEGV for address: 0x8
            Tue Mar  5 16:34:40 PST 2019: /home/kbuilder/go/src/pproftest/pprof-node6-2019-03-05-16-33-36-130163-0800/busybench/node_modules/segfault-handler/build/Release/segfault-handler.node(+0x2c99)[0x7f4fee9f0c99]
            Tue Mar  5 16:34:40 PST 2019: /lib/x86_64-linux-gnu/libpthread.so.0(+0x10330)[0x7f4ff11d6330]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal13GlobalHandles6CreateEPNS0_6ObjectE+0xed)[0xfa89cd]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v82V818GlobalizeReferenceEPNS_8internal7IsolateEPPNS1_6ObjectE+0x60)[0xb91840]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal20SamplingHeapProfiler12SampleObjectEPhm+0xee)[0x11b04ae]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal8NewSpace20InlineAllocationStepEPhS2_S2_m+0x86)[0x1021886]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal8NewSpace16EnsureAllocationEiNS0_19AllocationAlignmentE+0xc3)[0x1021ab3]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal4Heap11AllocateRawEiNS0_15AllocationSpaceENS0_19AllocationAlignmentE+0x130)[0xfc1440]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal4Heap20AllocateFillerObjectEibNS0_15AllocationSpaceE+0x1f)[0xfd39ef]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal7Factory15NewFillerObjectEibNS0_15AllocationSpaceE+0x2d)[0xf7a6dd]
            Tue Mar  5 16:34:40 PST 2019: node(_ZN2v88internal26Runtime_AllocateInNewSpaceEiPPNS0_6ObjectEPNS0_7IsolateE+0x60)[0x1203b30]
            Tue Mar  5 16:34:40 PST 2019: [0x1cfa823840dd]
            Tue Mar  5 16:34:40 PST 2019: Segmentation fault (core dumped)
```

I'm trying to reproduce on a GCP Ubuntu VM right now.

**Comment by nolanmar511** [comment]
Turns out Node 8.4 was being using, and I had a problem with how I was installing and selecting the node version.

This version has known issues. We prevent the profiler from being used with this version of node in cloud-profiler-nodejs (see [here](https://github.com/googleapis/cloud-profiler-nodejs/blob/84fa342bc5f470ff2ad5961cadea6e69e0875bb0/package.json#L98)). So, this is expected.

**Comment by nolanmar511** [comment]
Tests are passing for 8/10/11. I'm investigating 6.

**Comment by nolanmar511** [comment]
Looks like 6 was failing because it was the first test case running, and nvm wasn't installed until after the first bash script that was run exited. 

Fixed by moving installation of nvm into the script which called the test.

**Comment by nolanmar511** [comment]
I'm working on making the test output more verbose.

**Comment by aalexand** [comment]
@nolanmar511 should @ofrobots also review?

File: ts/src/profile-encoder.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
```
Comment (aalexand) [inline ts/src/profile-encoder.ts]: 2019?

File: ts/src/index.ts [hunk 2]
```diff
@@ -21,6 +21,7 @@ import {perftools} from '../../proto/profile';
 import * as heapProfiler from './heap-profiler';
 import * as timeProfiler from './time-profiler';
 
+export {encode} from './profile-encoder';
```
Comment (aalexand) [inline ts/src/index.ts]: Feels like this change could be a separate PR.

File: ts/src/profile-encoder.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import * as pify from 'pify';
+import {gzip as gzipPromise} from 'zlib';
+
+import {perftools} from '../../proto/profile';
+
+const gzip = pify(gzipPromise);
+
+export async function encode(profile: perftools.profiles.IProfile):
+    Promise<Buffer> {
+  const buffer = perftools.profiles.Profile.encode(profile).finish();
```
Comment (ofrobots) [inline ts/src/profile-encoder.ts]: A sync version of similar code also exists in `index.ts`. It would be good to move that to this file now that we have a file specifically called `profile-encoder.ts`.

File: ts/src/index.ts (hunk 2 above)
Comment (ofrobots) [reply ts/src/index.ts]: The new symbol is being used from the tests, so it could belong in this PR.

File: .kokoro/continuous/system_test.cfg [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
[the remark this answers is not in the document]
Comment (nolanmar511) [reply .kokoro/continuous/system_test.cfg]: The unit tests for Node.js modules under googleapis are there. But, I agree.

I might do a follow-up change to re-organize kokoro configs at some point.

File: system-test/system_test.go [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+# redirected.
+. ~/.nvm/nvm.sh &>/dev/null # load nvm.
+{{if .NVMMirror}}NVM_NODEJS_ORG_MIRROR={{.NVMMirror}}{{end}} retry nvm install {{.NodeVersion}} &>/dev/null
+
+NODEDIR=$(dirname $(dirname $(which node)))
+
+# Build and pack pprof module.
+cd {{.PprofDir}}
+
+# TODO: remove this workaround.
```
[the remark this answers is not in the document]
Comment (nolanmar511) [reply system-test/system_test.go]: Done.

File: system-test/system_test.go [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+				profilePath := fmt.Sprintf("%s/busybench/%s.pb.gz", tc.name, wantProfile.profileType)
+				if err := checkProfile(profilePath, wantProfile); err != nil {
+					t.Errorf("failed to collect expected %s profile: %v", wantProfile.profileType, err)
+				}
+			}
+		})
+	}
+}
+
+func checkProfile(profilePath string, wantProfile profileSummary) error {
```
[the remark this answers is not in the document]
Comment (nolanmar511) [reply system-test/system_test.go]: Done.

File: ts/src/profile-encoder.ts (hunk 1 above)
Comment (nolanmar511) [reply ts/src/profile-encoder.ts]: Done.

File: ts/src/index.ts (hunk 2 above)
Comment (nolanmar511) [reply ts/src/index.ts]: I have https://github.com/google/pprof-nodejs/pull/14 for this now.
I can refactor on top of it.

File: ts/src/profile-encoder.ts (hunk 3 above)
Comment (nolanmar511) [reply ts/src/profile-encoder.ts]: https://github.com/google/pprof-nodejs/pull/14

File: system-test/busybench/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "busybench",
+  "version": "0.0.0",
```
[the remark this answers is not in the document]
Comment (nolanmar511) [reply system-test/busybench/package.json]: I switched to 1.0.0, because `npm init` automatically sets the version to 1.0.0.

The version here isn't really used (since this isn't an npm module).

**Comment by nolanmar511** [comment]
PTAL

**Comment by nolanmar511** [comment]
I can add a system-test to package.json, but the current system-test fails when called with "npm run system-test" (where the system-test script is "sh system-test/system_test.sh".

It seems like nvm doesn't really work well when called from "npm run".

**Comment by ofrobots** [comment]
In that case it would be good to add a comment in the README on how to run the system tests.

**Comment by nolanmar511** [comment]
Added a section to the README about running the system test.

PTAL.
